digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/lane/LaneFiller.scala

```diff
@@ -171,8 +173,27 @@ class LaneFiller {
     (adjustedAsset, mValueAdjustments)
   }
 
+  // Similar as function in LaneUtils. It's duplicate because we can't call LaneUtils here
+  def fixSideCode( roadAddress: BasicRoadAddress , laneCode: String ): SideCode = {
+    roadAddress.track match {
+      case 1 | 2 => SideCode.BothDirections // This means the road may have both ways with something between them (like highways or similar)
+      // The representation of the lane will be in the middle 'of the road'
+
+      case _ => roadAddress.sideCode match { // In this case the road have both ways 'connected' so we need to take attention of the SideCode and laneCode
+                                             // This will have influence in representation of the lane
+                              case SideCode.AgainstDigitizing => if (laneCode.startsWith("1")) SideCode.AgainstDigitizing
+                                                                  else SideCode.TowardsDigitizing
+
+                              case SideCode.TowardsDigitizing => if (laneCode.startsWith("1")) SideCode.TowardsDigitizing
+                                                                  else SideCode.AgainstDigitizing
+
+                              case _ => SideCode.BothDirections
+                            }
+      }
+
+  }
```

Comment:
Please check the current method "getSideCode" existent at "TierekisteriImporterOperations" file.

We already perform something very similar to this when we import multiple assets from Tierekisteri using that information (RoadAddress Information).

Saying this I think we need to modify this code to something similar to:

```Scala
  def fixSideCode(roadAddress: BasicRoadAddress, laneCode: String): SideCode = {
    Track.apply(roadAddress.track) match {
      case Track.RightSide =>
        roadAddress.sideCode
      case Track.LeftSide =>
        roadAddress.sideCode match {
          case SideCode.TowardsDigitizing => SideCode.AgainstDigitizing
          case SideCode.AgainstDigitizing => SideCode.TowardsDigitizing
          case _ => SideCode.BothDirections
        }
      case _ =>
        SideCode.BothDirections
    }
  }

Reply:
The logic inside TierekisteriImporterOperations is different.
For lanes we need a different logic.

Reply:
Ok, with that information seems fine.